config/config.go

```diff
@@ -664,6 +671,43 @@ func (c *KubernetesSDConfig) UnmarshalYAML(unmarshal func(interface{}) error) er
 	return checkOverflow(c.XXX, "kubernetes_sd_config")
 }
 
+// EC2SDConfig is the configuration for EC2 based service discovery.
+type EC2SDConfig struct {
+	Region          string                    `yaml:"region"`
+	Access          string                    `yaml:"access,omitempty"`
+	Secret          string                    `yaml:"secret,omitempty"`
+	RefreshInterval Duration                  `yaml:"refresh_interval,omitempty"`
+	Port            int                       `yaml:"port"`
+	Tags            map[string]string         `yaml:"tags,omitempty"`
+	RegexTags       map[string]*regexp.Regexp `yaml:"-"`
+	Public          bool                      `yaml:"public,omitempty"`
+	// Catches all undefined fields and must be empty after parsing.
+	XXX map[string]interface{} `yaml:",inline"`
+}
+
+// UnmarshalYAML implements the yaml.Unmarshaler interface.
+func (c *EC2SDConfig) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	*c = DefaultEC2SDConfig
+	type plain EC2SDConfig
+	err := unmarshal((*plain)(c))
+	if err != nil {
+		return err
+	}
+	if c.Port == 0 {
```

Comment:
I've changed my mind and now think it's best to default this to 80. This is because in the case where a user wants to use the public ip address via relabeling, the port is unused. Hence, I think we should make it optional (i.e., default to 80) so that it need not be specified when the public ip is used.